### PR TITLE
Include module name and type in validation error messages

### PR DESCRIPTION
**Pull request by RohithVangalla1**
Previously, validation errors from ModuleValidator.validate() did not indicate which specific sub-module caused the failure. For models with many layers, this made debugging DP compatibility issues difficult — users had to manually inspect each layer to find the offending module.

This change prepends the module path and type to each validation error message. For example, instead of:
  'BatchNorm cannot support DP'
users now see:
  'encoder.layer.3.norm (BatchNorm2d): BatchNorm cannot support DP'

This makes it immediately clear which layer needs to be replaced, especially in large models with hundreds of sub-modules.

Addresses the TODO:
  'use module name here - it's useful part of error message'

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs change / refactoring / dependency upgrade

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the **CONTRIBUTING** document and completed the CLA (see **CONTRIBUTING**).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

**Comment by iden-kalemaj** [comment]
@RohithVangalla1  thank you for this PR, can you please include a small unit test that verifies the desired error message output?

**Comment by RohithVangalla1** [comment]
Hi @iden-kalemaj , Thank you so much for the review: 

Added two unit tests in `a2ab7e5`:
- `test_validate_error_includes_module_name`: A simple model with `nn.BatchNorm1d` named `"bn"` — verifies the error message contains both `"bn"` and `"BatchNorm1d"`
- `test_validate_error_includes_nested_module_name`: A nested model with `block.norm` path — verifies the error contains the full dotted path `"block.norm"` and `"BatchNorm1d"`

Example of the new error output:
bn (BatchNorm1d): Batch normalization cannot support DP ...
block.norm (BatchNorm1d): Batch normalization cannot support DP ...
